### PR TITLE
add support for dcos-log v2

### PR DESCRIPTION
**Pull request by mnaboka**
affected subcommands:
- [X] `dcos task <name> log`
- [X]  `dcos node log ...`
- [X] `dcos service log <name>`
- [X] unittests

If DC/OS cluster has the `LOGGING_V2` capability, dcos-cli should use [dcos-log v2](https://docs.google.com/document/d/1Z2gWi1H7lc3nNGqLUU9Hecvm9EyPLI-6t3uK1ANlHyA/edit#), otherwise based on the logging strategy mesos files API or dcos-log v1 will be used.

### CODE REVIEW

**Comment by mnaboka** [comment]
run linux integration tests

**Comment by mnaboka** [comment]
run linux integration tests

File: cli/dcoscli/node/main.py [hunk 1]
```diff
@@ -714,31 +718,90 @@ def _get_unit_type(unit_name):
...
 
     :param component: DC/OS component name
     :type component: str
+    :param version: Use logging API version. Default to 1.
+    :rtype: int
     :return: leader logs url
     :rtype: str
     """
 
+    if version < 1 and version > 2:
```
Comment (bamarni) [inline cli/dcoscli/node/main.py]: this can't be evaluated to true, did you mean `or`?

File: cli/dcoscli/task/main.py [hunk 2]
```diff
@@ -336,6 +341,45 @@ def get_nested_container_id(task):
...
+    :type task: str
+    :param lines: number of lines to print
+    :type lines: int
+    :param file_: file path to read
+    :type file_: str
+    """
+
+    if len(tasks) != 1:
+        raise DCOSException(
+            "found more then one task with the same name: {}. Please provide "
```
Comment (bamarni) [inline cli/dcoscli/task/main.py]: `then` -> `than`

File: cli/dcoscli/task/main.py [hunk 3]
```diff
@@ -336,6 +341,45 @@ def get_nested_container_id(task):
     return '.'.join(reversed(container_ids))
 
 
+def _dcos_log_v2(follow, tasks, lines, file_):
+    """ a client to dcos-log v2
+
+    :param follow: same as unix tail's -f
+    :type follow: bool
+    :param task: task pattern to match
+    :type task: str
```
Comment (bamarni) [inline cli/dcoscli/task/main.py]: the param name is `tasks`

File: cli/tests/unit/test_node.py [hunk 4]
```diff
@@ -191,6 +191,87 @@ def test_node_diagnostics_download(mock_get_diagnostics_list, mock_do_request,
         stream=True)
 
 
+@mock.patch('dcos.config.get_config_val')
+@mock.patch('dcos.http.get')
+@mock.patch('dcoscli.log.dcos_log_enabled')
+def test_dcos_log_v2_leader_mesos(mocked_dcos_log_enabked, mocked_http_get,
```
Comment (bamarni) [inline cli/tests/unit/test_node.py]: there are multiple occurences of `mocked_dcos_log_enabked` instead of `mocked_dcos_log_enabled` in this file.

**Comment by mnaboka** [comment]
run linux integration tests

**Comment by mnaboka** [comment]
@bamarni i addressed your comments, please review again

**Pull request merged**